did/ion/did.go

```diff
@@ -85,3 +108,216 @@ func LongToShortFormDID(longFormDID string) (string, error) {
 	}
 	return shortFormDID, nil
 }
+
+// PatchesToDIDDocument applies a list of sidetree state patches in order resulting in a DID Document.
+func PatchesToDIDDocument(shortFormDID, longFormDID string, patches []any) (*did.Document, error) {
+	if len(patches) == 0 {
+		return nil, errors.New("no patches to apply")
+	}
+	if shortFormDID == "" {
+		return nil, errors.New("short form DID is required")
+	}
+	doc := did.Document{
+		Context:     []string{"https://www.w3.org/ns/did/v1"},
+		ID:          shortFormDID,
+		AlsoKnownAs: longFormDID,
+	}
+	for _, patch := range patches {
+		knownPatch, err := tryCastPatch(patch)
+		if err != nil {
+			return nil, err
+		}
+		switch knownPatch.(type) {
+		case AddServicesAction:
+			addServicePatch := knownPatch.(AddServicesAction)
+			doc.Services = append(doc.Services, addServicePatch.Services...)
```

Comment:
If you do this, then you don't need to re-cast within each `case`. You can use `typedPatch`
```suggestion
		switch typedPatch := knownPatch.(type) {
		case AddServicesAction:
			doc.Services = append(doc.Services, typedPatch.Services...)
```